enhancements/machine-config/bootc_update_path.md

```diff
@@ -0,0 +1,185 @@
+
+### Workflow Description
+
+1. The cluster administrator wants to specify osImageURL Update / Kernel Argument Update / Extension Change / Kernel Type Switch to the OS on each of their cluster nodes 
+2. The cluster administrator creates new Machine Configs for the changes they want to apply and deploys them to the cluster 
+3. The Machine Config Daemon will read the desired change in the new config
+
+| Type of update | OCL enabled? | Action |
+| -------------  | ------------ | ------ |
+| osImageURL Update | N | Enter the updateLayeredOS update path, where the system OS will be updated to the one specified in the new config by calling “bootc -switch …” |
+| osImageURL Update | Y | Enter the updateClusterBuild update path, where the new changes will be built into a new image which will be rolled out onto the node |
+| Kernel Argument Update | N | Enter the applyOSChanges update path, where the kernel args will be adjusted when new args are injected. The changes will be deployed by running “rpm-ostree kargs …” |
+| Kernel Argument Update | Y | Enter the updateClusterBuild update path, where the new changes will be built into a new image which will be rolled out onto the node |
+| Extension Change | N | Send out warning logs to notice the cluster admin that the change can not applied with bootc TP on but OCL off |
+| Extension Change | Y | Enter the updateClusterBuild update path, where the new changes will be built into a new image which will be rolled out onto the node |
+| Kernel Type Switch | N | Send out warning logs to notice the cluster admin that the change can not applied with bootc TP on but OCL off |
+| Kernel Type Switch | Y | Enter the updateClusterBuild update path, where the new changes will be built into a new image which will be rolled out onto the node |
+
+
+
+| zebra stripes | are neat      |    $1 |
+
+### API Extensions
+
+We don’t anticipate any user facing API extension changes as of now. We will update this section if something comes up during the smoke tests or implementation.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
```

Comment:
I think we will have to align with Hypershift in the future. As it stands Hypershift wouldn't utilize this codepath

Reply:
Hey, Jerry! Do you by any chance know if anyone from the Hypershift team would be interested in this as well? We could include him/her/them into this conversation and review process to get better alignment in early stages.